common/pinot/pinotQueryValidator.go

```diff
@@ -71,7 +71,7 @@ func (qv *VisibilityQueryValidator) ValidateQuery(whereClause string) (string, e
 
 		stmt, err := sqlparser.Parse(placeholderQuery)
 		if err != nil {
-			return "", &types.BadRequestError{Message: "Invalid query."}
+			return "", &types.BadRequestError{Message: "Invalid query." + err.Error()}
```

Comment:
Print error message here for a better debugging experience.

Reply:
I'm unsure if it is a good idea since this error will be propagated to the customers. Do we want to expose which DB or which tables we use for customers? We can log it for debugging purposes while exposing only a generic message to the client.
Or, we can extract details about which arguments in the query are invalid to help us understand what is wrong. See *json.UnmarshalTypeError is an example that can point to the exact failure while not exposing the whole thing.

Reply:
It's actually a sql expression parsing error. 
It will have something like:
"Invalid query.syntax error at position 53 near 'select'"
"Invalid query.syntax error at position 38 near 'sql'"
Doesn't expose DB details or tables.

Reply:
Let's add a space after the dot or change it to colon. query.syntax is a bit confusing